presto-twitter-functions/src/main/java/com/facebook/presto/twitter/functions/SnowflakeFunctions.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.twitter.functions;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+/* size in
+ * bits:  1                     41                      5     5        12
+ *       +-+-----------------------------------------+-----+-----+------------+
+ *       |0|              milliseconds               |clstr|instc|  sequence  |
+ *       |0|              since twepoch              |  id |  id |   number   |
+ *       +-+-----------------------------------------+-----+-----+------------+
+ *        |
+ *        +- Most significant bit
+ */
+public class SnowflakeFunctions
+{
+    private static final long SequenceNumBits = 12L;
+    private static final long MaxSequenceNum = (1L << SequenceNumBits) - 1;
+    private static final long SequenceNumMask = MaxSequenceNum;
+
+    private static final long InstanceIdBits = 5L;
+    private static final long MaxInstanceId = (1L << InstanceIdBits) - 1;
+    private static final long InstanceIdShift = 12L;
+    private static final long InstanceIdMask = MaxInstanceId << InstanceIdShift;
+
+    private static final long ClusterIdBits = 5L;
+    private static final long MaxClusterId = (1L << ClusterIdBits) - 1;
+    private static final long ClusterIdShift = InstanceIdShift + InstanceIdBits;
+    private static final long ClusterIdMask = MaxClusterId << ClusterIdShift;
+
+    private static final long TimestampBits = 41L;
+    private static final long MaxTimestamp = (1L << TimestampBits) - 1;
+    private static final long TimestampShift = ClusterIdShift + ClusterIdBits;
+    private static final long TimestampMask = MaxTimestamp << TimestampShift;
+
```

Comment:
Also add a static check to make sure all bits added up to 64, e.g., assert `(SequenceNumBits  + InstanceIdBits  + ClusterIdBits + TimestampBits +1) == 64`